### ISSUE TITLE
Publish images to GitHub Container Registry

### ISSUE THREAD

**Issue by YOU54F** [pact-foundation/devrel#29]
In order to avoid users who may be affected by rate limiting pulling images from DockerHub, it would be prudent to provide images published to [GHCR](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry)

Dockerhub profile :- https://hub.docker.com/u/pactfoundation

## Images / Repos and Build locations

- [ ] https://hub.docker.com/r/pactfoundation/pact-ref-verifier
    - https://github.com/pact-foundation/pact-reference/blob/master/rust/pact_verifier_cli/release-docker.sh
- [ ] https://hub.docker.com/r/pactfoundation/pact-ref-mock-server
    - https://github.com/pact-foundation/pact-reference/blob/master/rust/pact_mock_server_cli/release-docker.sh
- [ ] https://hub.docker.com/r/pactfoundation/rust-musl-build
    - https://github.com/pact-foundation/pact-reference/blob/master/rust/Dockerfile.musl-build
- [ ] https://hub.docker.com/r/pactfoundation/pact-stub-server
    - https://github.com/pact-foundation/pact-stub-server
- [ ] https://hub.docker.com/r/pactfoundation/pact-js-musl
    - ?
- [ ] https://hub.docker.com/r/pactfoundation/pact-mock-service
    - ?
- [ ] https://hub.docker.com/r/pactfoundation/pact-broker
    - https://github.com/pact-foundation/pact-broker-docker/blob/master/script/release-workflow/docker-push.sh
- [ ] https://hub.docker.com/r/pactfoundation/pact-cli
    - https://github.com/pact-foundation/pact-ruby-cli/blob/master/script/release-workflow/docker-push.sh

## Delta of required changes

https://github.com/pact-foundation/pact-ruby-cli/pull/127/files

## Progress

- [ ] [Pact-Ruby-CLI](https://github.com/pact-foundation/pact-ruby-cli/pull/127)

**Comment by JP-Ellis** [comment]
Using GitHub's container registry is great.

It would be good to document the change from an end-user perspective. If I used to `docker run --rm pactfoundation/foo`, what do I run now?

**Comment by YOU54F** [comment]
# Pact Stub Server

Now complete

- [Action Run](https://github.com/pact-foundation/pact-stub-server/actions/runs/9034439839/job/24827509574)
- [Release](https://github.com/pact-foundation/pact-stub-server/releases/tag/v0.6.0)
- [DockerHub Image](https://hub.docker.com/r/pactfoundation/pact-stub-server)
- [GHCR Image](https://github.com/pact-foundation/pact-stub-server/pkgs/container/pact-stub-server)

## DockerHub

```
docker pull pactfoundation/pact-stub-server
docker run -t -p 8080:8080 -v "$(pwd)/pacts/:/app/pacts" pactfoundation/pact-stub-server -p 8080 -d pacts
```

## Github Container Registry

```
docker pull ghcr.io/pact-foundation/pact-stub-server
docker run -t -p 8080:8080 -v "$(pwd)/pacts/:/app/pacts" ghcr.io/pact-foundation/pact-stub-server -p 8080 -d pacts
```

**Comment by mjpieters** [comment]
Can this also be implemented for the pact-ref-verifier image from the pact-reference repository? I noticed that recent versions from that repo haven't been pushed to any registry (see pact-foundation/pact-reference#446).

**Comment by YOU54F** [comment]
yep sorry haven't got round to adding an auto publish to docker for the verifier and mock server  cli. will look to get in place possibly next week
